src/main/java/org/testng/TestNG.java

```diff
@@ -211,11 +213,22 @@ private void init(boolean useDefaultListeners) {
   }
 
   public int getStatus() {
+    if (m_status == HAS_STARTED) {
+      //we perhaps are running in the JUnit mode. So the ExitCodeListener is not involved.
+      //Lets just override the status and return it as 0
+      return SUCCESS;
+    }
     return m_status;
   }
 
   private void setStatus(int status) {
-    m_status |= status;
+    if (m_status == HAS_STARTED || status == SUCCESS) {
```

Comment:
If some tests are not success but the last one is success, then the testng status will be success, right?
Is it the expected behavior?

Reply:
The test case should be added. We can enforce a successful test with `priority`.